arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,64 +1,86 @@
+## Retryable Tickets Lifecycle
 
+Here we walk through the different stages of the lifecycle of a retryable ticket, shown in the diagram below.
+
+<RetryablesLifeCycleDiagram />
+
+1. Creating a retryable ticket is initiated with a transaction call to the `createRetryableTicket` function of the `inbox` contract. This function has a set of parameters which lead to a succesfull creation (and redemption) of the ticket if set carefully:
+
+    - `address to`: The destination L2 contract address
+    - `uint256 l2CallValue`: The callvalue for retryable L2 message
+    - `uint256 maxSubmissionCost`: The maximum amount of gas to be paid for submitting the ticket. This amount is deducted from sender's L2 balance and is directly proportional to the size of the retryable’s data  
+    - `address excessFeeRefundAddress`: The L2 address to which the excess fee is creadited (gasLimit * maxFeePerGas - ticket execution cost)
+    - `address callValueRefundAddress`: The L2 address to which the l2Callvalue if the ticket times out or gets cancelled
+    - `uint256 gasLimit`: Maximum amount of  gas deducted from sender's L2 balance to cover L2 execution of the ticket
+    - `uint256 maxFeePerGas`: The Price bid for L2 execution of the ticket
+    - `bytes calldata data`: ABI encoded data of L2 message
+
+2. As shown in the diagram above, the sender's deposit must be enough to make the submission succeed (i.e, cover the L1 submission and L2 execution of the ticket).
+3. If provided correctly, a new ticket with a unique `TicketID` is created and added to retryable buffer. Also, funds (`submissionCost` + `l2CallValue`) are deducted from the sender and placed into the escrow for later use in redeeming the ticket.
+4. Ticket creation causes the [`ArbRetryableTx`](./precompiles#ArbRetryableTx) precompile to emit a `TicketCreated` event containing the `TicketID`.
+5. It is very important to note that the submission of a ticket on L1 is separable / asynchronous from its execution on L2, i.e., a successful L1 ticket creation does not guarantee a successful redemption. Assuming the sender has provided enough gas that can cover the L2 execution/redemption of the ticket on step 1,  ticket's submission is followed by an attempt to execute it on L2 (i.e., an **auto-redeem** using the supplied gas, as if the [`redeem`](./precompiles#ArbRetryableTx) method of the ArbRetryableTx precompile had been called). Depending on how much gas the sender has provided in step 1, ticket's redemption can either (1) immediately succeed or (2) fail. We explain both situations here:
```

Comment:
> Assuming the sender has provided enough gas that can cover the L2 execution/redemption of the ticket on step 1
As with above: a bit of a nuance, but the sender was _forced_ to provide enough gas to meet their gas price / value arguments; so this case is really just assuming that both maxGas and maxPrice are > 0

Reply:
You're right. I'd made it clear at the beginning of the section that _any_ submission is followed by an attempt for the redemption (though not guaranteed).That being said,  I'd say we could just remove `Assuming the sender has provided enough gas that can cover the L2 execution/redemption of the ticket on step 1` then. What's your call?